packages/cli/src/commands/experimental/generateRealtimeHandler.js

```diff
@@ -0,0 +1,270 @@
+import { Listr } from 'listr2'
+import pascalcase from 'pascalcase'
+import pluralize, { singular } from 'pluralize'
+import prompts from 'prompts'
+
+import { generate as generateTypes } from '@redwoodjs/internal/dist/generate/generate'
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import {
+  generateTemplate,
+  getPaths,
+  transformTSToJS,
+  writeFile,
+} from '../../lib'
+import c from '../../lib/colors'
+import { isTypeScriptProject } from '../../lib/project'
+
+import { command, description, EXPERIMENTAL_TOPIC_ID } from './setupRealtime'
+import { printTaskEpilogue, isServerFileSetup, isRealtimeSetup } from './util'
+
+const templateVariables = (name) => {
+  name = singular(name.toLowerCase())
+
+  return {
+    name,
+    collectionName: pluralize(name),
+    pluralName: pluralize(name),
+    pluralPascalName: pascalcase(pluralize(name)),
+    camelName: camelcase(name),
+    functionName: camelcase(name),
+    liveQueryName: `recent${pascalcase(pluralize(name))}`,
+    subscriptionQueryName: `recent${pascalcase(pluralize(name))}`,
+    subscriptionName: `listenTo${pascalcase(name)}Channel`,
+    modelName: pascalcase(name),
+    typeName: pascalcase(name),
+    channelName: `${pascalcase(name)}Channel`,
+    subscriptionInputType: `Publish${pascalcase(name)}Input`,
+    subscriptionServiceResolver: `publishTo${pascalcase(name)}Channel`,
+  }
+}
+
+export async function handler({ name, type, force, verbose }) {
+  const redwoodPaths = getPaths()
+  const ts = isTypeScriptProject()
+  name = singular(name.toLowerCase())
+
+  let functionType = type
+
+  // Prompt to select what type if not specified
+  if (!functionType) {
+    const response = await prompts({
+      type: 'select',
+      name: 'functionType',
+      choices: [
+        {
+          value: 'liveQuery',
+          title: 'Live Query',
+          description: 'Create a Live Query to watch for changes in data',
+        },
+        {
+          value: 'subscription',
+          title: 'Subscription',
+          description: 'Create a Subscription to watch for events',
+        },
+      ],
+      message: 'What type of realtime event would you like to create?',
+    })
+
+    functionType = response.functionType
+  }
+
+  const tasks = new Listr(
+    [
+      {
+        title: 'Confirmation',
+        task: async (_ctx, task) => {
+          const confirmation = await task.prompt({
+            type: 'Confirm',
+            message:
+              'Realtime is currently an experimental RedwoodJS feature. Continue?',
+          })
+
+          if (!confirmation) {
+            throw new Error('User aborted')
+          }
+        },
+      },
+      {
+        title: 'Checking for realtime environment prerequisites ...',
+        task: () => {
+          isServerFileSetup() && isRealtimeSetup()
+        },
+      },
+      {
+        title: `Adding ${name} example subscription ...`,
+        skip: () => functionType !== 'subscription',
```

Comment:
If you use `enabled` here in place of `skip` it won't even show the title for the type the user didn't choose.